File: android/src/main/java/com/shopify/rnandroidauto/CarService.java
package com.shopify.rnandroidauto;

import androidx.annotation.NonNull;
import androidx.car.app.CarAppService;
import androidx.car.app.Session;
import androidx.car.app.validation.HostValidator;

import com.facebook.react.ReactApplication;
import com.facebook.react.ReactInstanceManager;

public final class CarService extends CarAppService {
    private ReactInstanceManager mReactInstanceManager;

    @Override
    public void onCreate() {
        mReactInstanceManager = ((ReactApplication) getApplication()).getReactNativeHost().getReactInstanceManager();
    }

    @NonNull
    @Override
    public HostValidator createHostValidator() {
        return HostValidator.ALLOW_ALL_HOSTS_VALIDATOR;
    }

    @NonNull
    @Override
    public Session onCreateSession() {
        return new CarServiceSession(mReactInstanceManager.getCurrentReactContext());
    }
}

File: android/src/main/java/com/reactnativeandroidauto/CarService.kt
package com.reactnativeandroidauto

import androidx.car.app.CarAppService
import androidx.car.app.Session
import androidx.car.app.validation.HostValidator
import com.facebook.react.ReactApplication
import com.facebook.react.ReactInstanceManager

class CarService : CarAppService() {
  private lateinit var reactInstanceManager: ReactInstanceManager
  override fun onCreate() {
    reactInstanceManager =
      (application as ReactApplication).reactNativeHost.reactInstanceManager
  }

  override fun createHostValidator(): HostValidator {
    return HostValidator.ALLOW_ALL_HOSTS_VALIDATOR
  }

  override fun onCreateSession(): Session {
    return CarServiceSession(reactInstanceManager.currentReactContext)
  }
}
